src/linear_algebra/affine_space.lean

```diff
@@ -212,6 +233,77 @@ end
 
 end finset
 
+section affine_independent
+
+open add_torsor
+
+variables (k : Type*) (V : Type*) {P : Type*} [ring k] [add_comm_group V] [module k V]
+variables [affine_space k V P] {ι : Type*}
+
+/-- An indexed family is said to be affinely independent if no
+nontrivial weighted subtractions (where the sum of weights is 0) are
+0. -/
+def affine_independent (p : ι → P) : Prop :=
+∀ (s : finset ι) (w : ι → k) (h : ∑ i in s, w i = 0), s.weighted_vsub V p w = 0 → ∀ i ∈ s, w i = 0
+
+/-- A family with at most one point is affinely independent. -/
+lemma affine_independent_of_subsingleton [subsingleton ι] (p : ι → P) :
+  affine_independent k V p :=
+λ s w h hs i hi, fintype.eq_of_subsingleton_of_sum_eq h i hi
+
+/-- A family is affinely independent if and only if the differences
+from a base point in that family are linearly independent. -/
+lemma affine_independent_iff_linear_independent_vsub (p : ι → P) (i1 : ι) :
```

Comment:
Would this proof be easier if things like `weighted_vsub` used `finsupp` instead of a pair of `ι → k` and `s : finset ι`? Potentially there should be a definition of a free affine space, and affine independence should be defined in terms of this. Do we have morphisms of affine spaces? I don't have clear understanding of this though. @jcommelin any thoughts?

Reply:
I think it's natural for `weighted_vsub` to be defined in terms of sums over a `finset` since that's how all the lemmas in `big_operators` are defined. And just as it was concluded in #3122 not to have duplicate copies of the API for the case of summing over `finset.univ`, so it seems best not to duplicate it for sums over `finsupp.support` either. (I do expect to end up with at least one lemma involving `finsupp`: for all points in the affine span of an affine independent family, there exists a unique `finsupp` with sum 1 whose `affine_combination` is that point, i.e. a unique choice of barycentric coordinates for that point.)

Reply:
Yup, I don't like `finsupp.sum` etc... I would be in favour of removing that, because the API for `finset.sum` is a lot better.